signac/diff.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from .contrib.collection import _traverse_filter
+
+
+def _dotted_keys_to_nested_dicts(mapping):
+    """Converts dictionaries with dot-separated keys into nested dictionaries.
+
+    :param mapping: A mapping with dots in its keys, e.g. {'a.b': 'c'}
+    :returns: A mapping with nested keys, e.g. {'a': {'b': 'c'}}
+    """
+    result = {}
+
+    def make_nested_dict(d, keys):
+        item = d
+        for key in keys[:-1]:
+            if key not in item:
+                item[key] = {}
+            item = item[key]
+        return item
+
+    for dotted_key, value in mapping.items():
+        keys = dotted_key.split('.')
+        make_nested_dict(result, keys)[keys[-1]] = value
+
+    return result
+
+
+def diff_jobs(*jobs):
+    """Find differences among a list of jobs' state points.
+
+        :param jobs: One or more jobs whose state points will be diffed.
```

Comment:
Also the formatting of this `:param jobs:` implies that `jobs` is a specific argument, while we need it to be clear that it's a variable number of arguments.

This is somewhat difficult to make work correctly in the docs -- to the best of my recollection, the correct way to do it is:

```python
r"""Find differences among a list of jobs' state points.

The resulting diff is a dictionary where the keys are job ids and the values are each job's state point minus the intersection of all provided jobs' state points.

:param \*jobs: One or more jobs whose state points will be diffed.
:returns: A dictionary whose keys are job ids and whose values are the unique parts of that job's state point.
"""
```
Notice that we have to use a raw string with the `r"""` at the beginning, so that the backslash is interpreted as a literal backslash.